### ISSUE TITLE
Reliance on Templates integration?

### ISSUE THREAD

**Issue by The-Arsonist** [gfrancesco/cwww-template-light-ha#1]
I get an invalid configuration message when I load this into my configuration.yaml. Does this rely on the template/light integration, or is it standalone?

living_room_strip:
  friendly_name:        "Living Room Strip"
  value_template:       >-
                        {{ ( state_attr('light.light.strip_channel_2', 'brightness') | int(0) +
                             state_attr('light.light.strip_channel_1', 'brightness') | int(0) ) > 0 }}
  level_template:       "{{  states('input_number.LRS_LIGHT_TEMPERATURE')  | int(0)   }}"
  temperature_template: "{{  states('input_number.LRS_LIGHT_TEMPERATURE') | int(500) }}"
  set_level:
    - service:     input_number.set_value
      target:
        entity_id: input_number.LRS_LIGHT_TEMPERATURE
      data:
        value:     "{{ brightness | int(0) }}"
    - if:
        - condition: numeric_state
          entity_id: input_number.LRS_LIGHT_TEMPERATURE
          above:     128
      then:
        - variables:
            perc_above_mid: "{{ ((states('input_number.LRS_LIGHT_TEMPERATURE') | int(0) - 128) / 127) | float(0) }}"
        - if:
            - condition:    numeric_state
              entity_id:    input_number.LRS_LIGHT_TEMPERATURE
              above:        327
          then:
            - variables:
                warmest:          "{{ (500 - (perc_above_mid * 173)) | int(500) }}"
            - if:
                - condition:      numeric_state
                  entity_id:      input_number.LRS_LIGHT_TEMPERATURE
                  above:          0
                  value_template: "{{ (float(state.state) - warmest) | int(0) }}"
              then:
                - service:        input_number.set_value
                  target:
                    entity_id:    input_number.LRS_LIGHT_TEMPERATURE
                  data:
                    value:        "{{ warmest | int(500) }}"
        - if:
            - condition:    numeric_state
              entity_id:    input_number.LRS_LIGHT_TEMPERATURE
              below:        327
          then:
            - variables:
                coldest:          "{{ (153 + (perc_above_mid * 174)) | int(153) }}"
            - if:
                - condition:      numeric_state
                  entity_id:      input_number.LRS_LIGHT_TEMPERATURE
                  below:          0
                  value_template: "{{ (float(state.state) - coldest) | int(0) }}"
              then:
                - service:        input_number.set_value
                  target:
                    entity_id:    input_number.LRS_LIGHT_TEMPERATURE
                  data:
                    value:        "{{ coldest | int(153) }}"
    - service:     light.turn_on
      target:
        entity_id: light.living_room_strip
  set_temperature:
    - service:     input_number.set_value
      target:
        entity_id: input_number.LRS_LIGHT_TEMPERATURE
      data:
        value:     "{{ color_temp | int(500) }}"
    - if:
        - condition: numeric_state
          entity_id: input_number.LRS_LIGHT_TEMPERATURE
          above:     128
      then:
        - variables:
            delta_perc_mean_temp: "{{ (((( (states('input_number.LRS_LIGHT_TEMPERATURE') | int(500) - 152) / 348) - 0.5) | abs ) * 2) | float(0) }}"
            max_brightness:       "{{ ( (1 - delta_perc_mean_temp) * 127 + 128 ) | int(0) }}"
        - if:
            - condition:          numeric_state
              entity_id:          input_number.LRS_LIGHT_TEMPERATURE
              above:              0
              value_template:     "{{ (float(state.state) - max_brightness) | int(0) }}"
          then:
            - service:            input_number.set_value
              target:
                entity_id:        input_number.LRS_LIGHT_TEMPERATURE
              data:
                value:            "{{ max_brightness | int(0) }}"
    - service:     light.turn_on
      target:
        entity_id: light.living_room_strip
  turn_on:
    - variables:
        ww_ratio:   "{{ ((   states('input_number.LRS_LIGHT_TEMPERATURE') | int(500) - 152 ) / 348 ) | float(0) }}"
    - service:      light.turn_on
      target:
        entity_id:  light.light.strip_channel_2
      data:
        brightness: "{{ (2 * states('input_number.LRS_LIGHT_TEMPERATURE') | int(0) * ww_ratio)        | int(0) }}"
    - service:      light.turn_on
      target:
        entity_id:  light.light.strip_channel_1
      data:
        brightness: "{{ (2 * states('input_number.LRS_LIGHT_TEMPERATURE') | int(0) * (1 - ww_ratio) ) | int(0) }}"
  turn_off:
    - service:      light.turn_off
      target:
        entity_id:  light.light.strip_channel_2
    - service:      light.turn_off
      target:
        entity_id:  light.light.strip_channel_1

**Comment by gfrancesco** [comment]
What error did you get?
The code defines a new template light, so it needs to be appended to your existing templates section (if you already have any). See the [official documentation on template lights](https://www.home-assistant.io/integrations/light.template) if you are only using `configuration.yaml` to define your HA configuration.

In my installation I decided to split the configuration into several files, so this is the relevant part in my `configuration.yaml`:

``` Yaml
light:
  - platform:     template
    lights:       !include_dir_merge_named   template_lights/
```

Then I created a new directory `template_lights/` where I saved `template_light_cwww.yaml` exactly as you find it in the repository.
